### PR TITLE
Fix shadow semaphores used in VkSubmitInfo

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

File: framework/decode/vulkan_replay_consumer_base.cpp [hunk 1]
```diff
@@ -7550,12 +7559,21 @@ void VulkanReplayConsumerBase::GetShadowSemaphores(
...
         {
             SemaphoreInfo* semaphore_info = object_info_table_.GetSemaphoreInfo(semaphore_infos[i].semaphore);
-            if ((semaphore_info != nullptr) && (semaphore_info->shadow_signaled == true))
+            if (semaphore_info != nullptr)
             {
-                // If found, unsignal the semaphore to represent it being used.
-                shadow_semaphores->push_back(semaphore_info);
-                semaphore_info->shadow_signaled = false;
-                shadow_semaphores_.erase(semaphore_info->handle);
+                if (semaphore_info->shadow_signaled == true)
```
Comment (MarkY-LunarG) [inline framework/decode/vulkan_replay_consumer_base.cpp:7564]: Is this getting complex enough that it (and the duplicate code above) should be in a helper function?

Comment (ziga-lunarg) [reply framework/decode/vulkan_replay_consumer_base.cpp]: Done

**Review by MarkY-LunarG — APPROVED** [review]
LGTM

**Comment by bradgrantham-lunarg** [comment]
Hey, @davidlunarg since you've been thinking about semaphores could you take a look at this?  Thank you!

**Comment by davidlunarg** [comment]
Sorry for the late response. I didn't see this back in May, and received the reminder last week while on vacation.

LGTM

**Pull request merged**